### PR TITLE
Custom filter

### PR DESCRIPTION
**Pull request by pgaglani1**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -381,3 +382,27 @@ For example, if the source bundle file is `com/ibm/g11n/en/MyMessages.properties
...
+
+
+# <a name="TOC-CustomFilter"></a>Custom Resource Filter
+
+[Globalization Pipeline Resource Filter](https://github.com/IBM-Cloud/gp-java-tools/tree/master/gp-res-filter) provides
+a mechanism to implement your own custom filter, that can be used through
+Globalization Pipeline Maven Plugin/Ant Task and other tools (including this Jenkins Plugin).
+
+To embed your own custom filter with this Jenkins plugin, follow these steps
+
```
Comment (srl295) [inline README.md]: 0. clone `IBM-Cloud/gp-jenkins` ?

File: README.md [hunk 2]
```diff
@@ -381,3 +382,27 @@ For example, if the source bundle file is `com/ibm/g11n/en/MyMessages.properties
...
+1. Add your custom resource filter dependency in [pom.xml](https://github.com/IBM-Cloud/gp-jenkins/blob/master/pom.xml)
+<img src="assets/gp-jenkins-csv-filter.png" alt="gp-jenkins-csv-filter"/>
```
Comment (srl295) [inline README.md]: images should show test un-commented:
```xml
<!-- this is an example --->
<dependency>
   … csv-filter
</dependency>
```

File: README.md (hunk 1 above)
Comment (srl295) [reply README.md]: need some level set, "to embed your own custom filter, you must compile a customer version of the Jenkins plugin…"?

File: src/main/java/org/jenkinsci/plugins/gpjenkins/GlobalizationPipelineBuilder.java [hunk 3]
```diff
@@ -856,7 +876,7 @@ else if(!workspace.child(outDir.trim()).isDirectory()){
 					}
 				}
 
-			} catch (ServiceException e) {
+			} catch (ServiceException | NullPointerException e) {
```
Comment (yumaoka) [inline src/main/java/org/jenkinsci/plugins/gpjenkins/GlobalizationPipelineBuilder.java]: Why do you need to catch NPE here? In general, this is not a good practice.

File: src/main/java/org/jenkinsci/plugins/gpjenkins/GlobalizationPipelineBuilder.java [hunk 4]
```diff
@@ -806,8 +808,26 @@ else if(!workspace.child(outDir.trim()).isDirectory()){
...
+					
+					
+					// UNCOMMENT FOR CUSTOM FILTER IMPLEMENTATION
+					/*
+					Thread.currentThread().setContextClassLoader(orig); 
+					*/
+					
+					
+					
+
```
Comment (yumaoka) [inline src/main/java/org/jenkinsci/plugins/gpjenkins/GlobalizationPipelineBuilder.java]: Two comments -

1. You don't need to comment out the code block by default. The code should not be harmful.
2. I can add filter method which takes class loader explicitly. But doing so, you don't need to set uberClassLoader to context classloader, and restore.

Comment (pgaglani1) [reply src/main/java/org/jenkinsci/plugins/gpjenkins/GlobalizationPipelineBuilder.java]: Removed comment block.. just showing real example with csv filter

File: README.md (hunk 2 above)
Comment (pgaglani1) [reply README.md]: Updated with real csv filter example

**Pull request merged**